client/src/controllers/AdminFileController.ts

```diff
@@ -1,13 +1,74 @@
-const apiURL = '/api/file/instructions';
+import { Survey } from '@interfaces/survey';
+import { request } from '@src/utils/request';
+
+const apiURL = '/api/file';
 
 export async function getInstructionFilename() {
-  const response = await fetch(apiURL, { method: 'HEAD' });
+  const response = await fetch(`${apiURL}/instructions`, { method: 'HEAD' });
   return JSON.parse(response.headers.get('File-details')).name;
 }
 
 export async function storeAdminInstructions(file: File) {
   const formData = new FormData();
   formData.append('file', file);
 
-  await fetch(apiURL, { method: 'POST', body: formData });
+  await fetch(`${apiURL}/instructions`, { method: 'POST', body: formData });
+}
+
+export async function duplicateFileOnDb(
+  originalPath: string,
+  surveyId: number,
+  organizationId: string,
+) {
+  const response = await request<{ url: string }>(
+    `${apiURL}/copy/${originalPath}`,
+    {
+      method: 'POST',
+      body: { surveyId, organizationId },
+    },
+  );
+  return response;
+}
+
+export async function duplicateFiles<T extends object>(
+  object: T,
+  activeSurvey: Survey,
+) {
+  if (typeof object !== 'object' || object == null) return object;
+  async function processFileUrl(
```

Comment:
[nit] I wouldn't nest these functions, it makes the parent function less readable at a glance... Also the types of the parameters `object` and `targetSurvey` should be defined.